syntax = "proto3";
option java_package = "com.grpc.demo.service.user";
option java_multiple_files = true;

package user;

message Empty {}

message User {
    int32 id = 1;
    string username = 2;
    string first_name = 3;
    string last_name = 4;
    string phone = 5;
    string email = 6;
    int32 role_id = 7;
    bool is_active = 8;
}

message Role {
    int32 id = 1;
    string name = 2;
}

message Response {
    bool success = 1;
    string message = 2;
}

message UserList {
    repeated User user = 1;
}

service UserService {
    rpc CreateUser(User) returns (Response);
    rpc UpdateUser(User) returns (Response);
    rpc DeleteUser(User) returns (Response);
    rpc ListUsers(Empty) returns (UserList);
}
